allennlp/commands/evaluate.py

```diff
@@ -123,47 +138,68 @@ def evaluate_from_args(args: argparse.Namespace) -> Dict[str, Any]:
 
-    evaluation_data_path = args.input_file
-    logger.info("Reading evaluation data from %s", evaluation_data_path)
-
-    data_loader_params = config.pop("validation_data_loader", None)
-    if data_loader_params is None:
-        data_loader_params = config.pop("data_loader")
-    if args.batch_size:
-        data_loader_params["batch_size"] = args.batch_size
-    data_loader = DataLoader.from_params(
-        params=data_loader_params, reader=dataset_reader, data_path=evaluation_data_path
-    )
-
-    embedding_sources = (
-        json.loads(args.embedding_sources_mapping) if args.embedding_sources_mapping else {}
-    )
+    # split files
+    evaluation_data_path_list = args.input_file.split(":")
+    if args.output_file is not None:
+        output_file_list = args.output_file.split(":")
+        assert len(output_file_list) == len(
+            evaluation_data_path_list
+        ), "number of output path must be equal number of dataset "
+    if args.predictions_output_file is not None:
+        predictions_output_file_list = args.predictions_output_file.split(":")
+        assert len(predictions_output_file_list) == len(
+            evaluation_data_path_list
+        ), "number of predictions_output_file path must be equal number of dataset "
+
+    # output file
+    output_file_path = None
+    predictions_output_file_path = None
+
+    for index in range(len(evaluation_data_path_list)):
+        config = deepcopy(archive.config)
+        evaluation_data_path = evaluation_data_path_list[index]
+        if args.output_file is not None:
+            output_file_path = output_file_list[index]
+        if args.predictions_output_file is not None:
+            predictions_output_file_path = predictions_output_file_list[index]
+
+        logger.info("Reading evaluation data from %s", evaluation_data_path)
+        data_loader_params = config.get("validation_data_loader", None)
+        if data_loader_params is None:
+            data_loader_params = config.get("data_loader")
+        if args.batch_size:
+            data_loader_params["batch_size"] = args.batch_size
+        data_loader = DataLoader.from_params(
+            params=data_loader_params, reader=dataset_reader, data_path=evaluation_data_path
+        )
 
-    if args.extend_vocab:
-        logger.info("Vocabulary is being extended with test instances.")
-        model.vocab.extend_from_instances(instances=data_loader.iter_instances())
-        model.extend_embedder_vocab(embedding_sources)
+        embedding_sources = (
+            json.loads(args.embedding_sources_mapping) if args.embedding_sources_mapping else {}
+        )
 
-    data_loader.index_with(model.vocab)
+        if args.extend_vocab:
+            logger.info("Vocabulary is being extended with test instances.")
+            model.vocab.extend_from_instances(instances=data_loader.iter_instances())
+            model.extend_embedder_vocab(embedding_sources)
```

Comment:
The embedding sources would be the same for each input file, if I understand correctly. We don't need to run this every time.

Reply:
I think we do. Yes, the embedding sources would be the same, but the extended vocab would be different.  (https://github.com/allenai/allennlp/blob/main/allennlp/models/model.py#L464)

Reply:
https://github.com/allenai/allennlp/pull/5340/commits/dfa612a9aa53a62bed344f0da0b1a5621cb469a4

This should resolve this problem.